connect/workflows/core.py

```diff
@@ -177,33 +181,35 @@ async def transmit(self) -> Optional[List[dict]]:
                             "headers": dict(post_result.headers.items()),
                         }
                     except Exception as ex:
-                        if isinstance(ex, httpx.ConnectTimeout) or isinstance(
-                            ex, httpx.ConnectError
+                        if (
+                            isinstance(ex, httpx.ConnectTimeout)
+                            or isinstance(ex, httpx.ConnectError)
+                            and self.do_retransmit
                         ):
-                            if self.do_retransmit:
-                                # send retransmit message to to Kafka to record
-                                # retransmit message contains only the URL that failed
-                                retransmit_message = self.message
-                                retransmit_message["target_endpoint_urls"] = [server]
-                                retransmit_message["status"] = "ERROR"
-                                kafka_producer = get_kafka_producer()
-                                await kafka_producer.produce(
-                                    "RETRANSMIT",
-                                    json.dumps(retransmit_message, cls=ConnectEncoder),
-                                )
-
-                                # publish retransmit message to NATS
-                                nats_client = await nats.get_nats_client()
-                                msg_str = json.dumps(
-                                    retransmit_message, cls=ConnectEncoder
-                                )
-                                await nats_client.publish(
-                                    nats_retransmit_subject, bytearray(msg_str, "utf-8")
-                                )
+                            # send retransmit message to to Kafka to record
```

Comment:
Do you think it would be helpful to encapsulate the retransmit process within a "private" function such as 

```
def _retransmit_message()
```
Or if we can "reuse it" we could have an "accessible" function.